### DISCUSSION TITLE
Changing user password works, but throws an error

### DISCUSSION THREAD

**Discussion by gizmo1-11** [keycloak/keycloak#19327] [Q&A]
I use keycloak 20.0.5 (docker) and configured a realm with ldap-user-federation (samba AD) successfulyl with TLS.
When an user changes his password in "account-console", it works, the password is successfully changed, but the user gets an error message "Could not modify attribute for DN [USEROU]" and the password-form doesn't disappear.
The keycloak-log says
WARN  [org.keycloak.events] (executor-thread-7) type=UPDATE_PASSWORD_ERROR, realmId=......, clientId=account-console, userId=.......:user, ipAddress=....., error=password_rejected, reason='Could not modify attribute for DN [USEROU]', auth_method=openid-connect, custom_required_action=UPDATE_PASSWORD, response_type=code, redirect_uri=https://server.domain/keycloak/realms/my_realm/account/#/security/signingin, remember_me=false, code_id=...., response_mode=fragment, username=user, authSessionParentId=..., authSessionTabId=.... 

regards
Th

**Comment by gizmo1-11** [comment]
Solved
It was not enough to delegate the right to change passwords to the LDAP-Proxy-User.
He needs also the right to write the field userAccountControl.

**Comment by Dennis1993** [comment]
The service account has useraccountcontrol permissions but the error appears. How can I see which attribute is a Problem?
The reset works. Or can I disable the error message?

**Reply by GeorgesBolssens-Toreon** [reply]
ldap:// or ldaps:// ? you can't reset a password over an unencrypted connection and not expect an error.

**Reply by Dennis1993** [reply]
I found the Problem yesterday.
I have created a mapper for an AD Attribute „employeeNumber“ and this field wasn‘t read only. So keycloak Trier to update these field on a password reset. The service user can‘t edit this attribute. I set the mapper to readonly and now it works! 😍

ldaps:// is Important too! 👍
Thanks for your help 🙏
